Soulgram.Interests/Soulgram.Interests.Application/Queries/GenGenresIdsByNameQuery.cs

```diff
@@ -0,0 +1,47 @@
+﻿using MediatR;
+using Soulgram.Interests.Application.Interfaces;
+using Soulgram.Interests.Domain;
+
+namespace Soulgram.Interests.Application.Queries;
+
+public class GenGenresIdsByNameQuery : IRequest<Dictionary<string, string?>?>
+{
+    public GenGenresIdsByNameQuery(ICollection<string>? names)
+    {
+        Names = names;
+    }
+
+    public ICollection<string>? Names { get; }
+}
+
+public class GenGenresIdsByNameQueryHandler : IRequestHandler<GenGenresIdsByNameQuery, Dictionary<string, string?>?>
+{
+    private readonly IRepository<Genre> _genreRepository;
+
+    public GenGenresIdsByNameQueryHandler(IRepository<Genre> genreRepository)
+    {
+        _genreRepository = genreRepository;
+    }
+
+    public async Task<Dictionary<string, string?>?> Handle(
+        GenGenresIdsByNameQuery command,
+        CancellationToken cancellationToken)
+    {
+        if (command.Names == null)
```

Comment:
Does it make sense to move it to some Validator? e.g class derived from FluentValidation.AbstractValidator

Reply:
sure, needed a task to add!